ui.frontend/src/view/FormFieldBase.js

```diff
@@ -396,10 +396,12 @@ class FormFieldBase extends FormField {
         if (this.widget) {
             this.toggle(required, "required");
             this.element.setAttribute(Constants.DATA_ATTRIBUTE_REQUIRED, required);
-            if (required === true) {
-                this.widget.setAttribute("required", "required");
-            } else {
-                this.widget.removeAttribute("required");
+            if(!(this.widget instanceof NodeList)) { // filtering radiobutton and checkboxGroup
```

Comment:
NodeList is a generic interface that applies to more than radiobutton or checkbox elements. Any component specific handling should be done in respective component views.

Reply:
Reason I kept it in top level was to reduce the code duplication for checkbox and radiobuttons. 

And when a component has multiple widgets which one do we set as required?  I believe we set it in parent div only so isn't this the generic behaviour?

Reply:
Think about a (custom) component say telephone number that capture country code and the number in a string and numeric/tel fields respectively. In this case both of the nodes in the parent div may be required to be marked required. FormFieldBase should offer any minimal things as it can't be changed by  customization.

Reply:
@ravise5 Can you move this logic specific to radio button and check box view layer to keep the base layer simple ?

Reply:
Okay.